adapters/bidder_test.go

```diff
@@ -0,0 +1,66 @@
+package adapters
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+func TestExtraRequestInfoConvertCurrency(t *testing.T) {
+	var (
+		givenValue float64 = 2
+		givenFrom  string  = "AAA"
+		givenTo    string  = "BBB"
+	)
+
+	testCases := []struct {
+		description   string
+		setMock       func(m *mock.Mock)
+		expectedValue float64
+		expectedError string
+	}{
+		{
+			description:   "Success",
+			setMock:       func(m *mock.Mock) { m.On("GetRate", "AAA", "BBB").Return(2.5, nil) },
+			expectedValue: 5,
+		},
+		{
+			description:   "Error",
+			setMock:       func(m *mock.Mock) { m.On("GetRate", "AAA", "BBB").Return(2.5, errors.New("some error")) },
+			expectedError: "some error",
+		},
+	}
+
+	for _, test := range testCases {
+		mockConversions := &mockConversions{}
+		test.setMock(&mockConversions.Mock)
+
+		extraRequestInfo := NewExtraRequestInfo(mockConversions)
+		result, err := extraRequestInfo.ConvertCurrency(givenValue, givenFrom, givenTo)
+
+		mockConversions.AssertExpectations(t)
+		if test.expectedError == "" {
+			assert.NoError(t, err, test.description+":err")
+			assert.Equal(t, test.expectedValue, result, test.description+":result")
+		} else {
+			assert.Error(t, err, test.description+":err")
+			assert.Empty(t, result, test.description+":result")
+		}
```

Comment:
For an easier read, we could probably remove the `if-else` clause altogether for two `assert.Equal()` calls instead.
```
44       testCases := []struct {
45           description   string
46           setMock       func(m *mock.Mock)
47           expectedValue float64
48 -         expectedError string
   +         expectedError error
49       }{
50           {
44               description:   "Success",
45               setMock:       func(m *mock.Mock) { m.On("GetRate", "AAA", "BBB").Return(2.5, nil) },
46               expectedValue: 5,
47           },
48           {
49               description:   "Error",
50               setMock:       func(m *mock.Mock) { m.On("GetRate", "AAA", "BBB").Return(2.5, errors.New("some error")) },
47 -             expectedError: "some error",
   +             expectedError: errors.New("The actual error message"),
48           },
49       }
                .
                .
                .
             result, err := extraRequestInfo.ConvertCurrency(givenValue, givenFrom, givenTo)

             mockConversions.AssertExpectations(t)
44 -         if test.expectedError == "" {
45 -             assert.NoError(t, err, test.description+":err")
46 -             assert.Equal(t, test.expectedValue, result, test.description+":result")
47 -         } else {
48 -             assert.Error(t, err, test.description+":err")
49 -             assert.Empty(t, result, test.description+":result")
50 -         }
   +             assert.Equal(t, test.expectedValue, result, test.description+":result")
   +             assert.Equal(t, test.expectedError, err, test.description+":result")
51       }
52   }
adapters/bidder_test.go
```